ssl/quic/quic_channel.c

```diff
@@ -526,6 +528,55 @@ SSL *ossl_quic_channel_get0_tls(QUIC_CHANNEL *ch)
     return ch->tls;
 }
 
+static void free_buf_mem(unsigned char *buf, size_t buf_len, void *arg)
+{
+    BUF_MEM_free((BUF_MEM *)arg);
+}
+
+int ossl_quic_channel_schedule_new_token(QUIC_CHANNEL *ch,
+                                         const unsigned char *token,
+                                         size_t token_len)
+{
+    int rc = 0;
+    QUIC_CFQ_ITEM *cfq_item;
+    WPACKET wpkt;
+    BUF_MEM *buf_mem = NULL;
+    size_t l = 0;
+
+    buf_mem = BUF_MEM_new();
+    if (buf_mem == NULL)
+        goto err;
+
+    if (!WPACKET_init(&wpkt, buf_mem))
+        goto err;
+
+    if (!ossl_quic_wire_encode_frame_new_token(&wpkt, token,
+                                               token_len)) {
+        WPACKET_cleanup(&wpkt);
+        goto err;
+    }
+
+    WPACKET_finish(&wpkt);
+
+    if (!WPACKET_get_total_written(&wpkt, &l))
+        goto err;
+
+    cfq_item = ossl_quic_cfq_add_frame(ch->cfq, 1,
+                                       QUIC_PN_SPACE_APP,
+                                       OSSL_QUIC_FRAME_TYPE_NEW_TOKEN, 0,
+                                       (unsigned char *)buf_mem->data, l,
+                                       free_buf_mem,
+                                       buf_mem);
```

Comment:
Do we need to keep hold of the `buf_mem` at this point? We can just pass `buf_mem->data`, and then subsequently set `buf_mem->data` to NULL and free `buf_mem`.